pkg/validators/retry.go

```diff
@@ -0,0 +1,265 @@
+package validators
```

Comment:
I would make it obvious in filename this is a middleware. Maybe at some point we will have a need for a separate `pkg/middlewares/`, or `pkg/validators/middleware` subpkg, but r/now maybe `retry_middleware.go` ?